### PR TITLE
add button to old EvaluationRowWidget to migrate submission quotas (SWC-5394)

### PR DESCRIPTION
**Pull request by zimingd**

### CODE REVIEW

**Comment by zimingd** [comment]
image
image
On button click:
image

On migration error (message from backend):
image
Realistically should not occur since the old UI w/ the migrate button will not even be shown if there is no quota on the Evaluation object. To get this message I had to manually change my local build to always show the old UI

**Comment by jay-hodgson** [comment]
@zimingd : This is all very scary UI (danger text, irreversible action where the outcome/consequences are not readily apparent).  I'm not sure if I would ever click this Migrate CTA with the UI proposed here.

File: src/main/java/org/sagebionetworks/web/client/widget/evaluation/EvaluationRowWidget.java [hunk 1]
```diff
@@ -75,10 +94,11 @@
 	}
 
 	@Inject
-	public EvaluationRowWidget(UserBadge userBadge, DateTimeUtils dateTimeUtils, SubmitToEvaluationWidget submitToEvaluationButton) {
+	public EvaluationRowWidget(UserBadge userBadge, DateTimeUtils dateTimeUtils, SubmitToEvaluationWidget submitToEvaluationButton, SynapseJavascriptClient synapseClient) {
 		this.userBadge = userBadge;
 		this.dateTimeUtils = dateTimeUtils;
 		this.submitToEvaluationButton = submitToEvaluationButton;
+		this.synapseClient = synapseClient;
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/evaluation/EvaluationRowWidget.java]: This is usually named "jsClient" in other areas of the code (different from the old GWT SynapseClient servlet RPCs)

File: src/main/java/org/sagebionetworks/web/client/widget/evaluation/EvaluationRowWidget.java [hunk 2]
```diff
@@ -131,6 +151,40 @@ public void configure(Evaluation evaluation, EvaluationActionHandler handler) {
...
+		modal.add(footer);
+
+		Button cancelButton = new Button("Cancel", (cancelClickEvent)->{modal.hide();});
+		footer.add(cancelButton);
+
+		Button migrate = new Button("Migrate", (migrateClickEvent)->{synapseClient.migrateSubmissionQuotaToEvaluationRound(evaluation.getId(),
+				new AsyncCallback<Void>() {
+					@Override
+					public void onFailure(Throwable throwable) {
+						body.add(new Alert("ERROR migrating evaluation quota: \n\n" +throwable.getMessage(), AlertType.DANGER));
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/evaluation/EvaluationRowWidget.java]: This error is never cleared.  One option is to inject a SynapseAlert and add it to the body.  Clear it before calling migrateSubmissionQuotaToEvaluationRound(), and on error tell it to "showError(throwable.getMessage())"

**Comment by zimingd** [comment]
@jay-hodgson 

image
image

File: src/main/java/org/sagebionetworks/web/client/widget/evaluation/AdministerEvaluationsList.java [hunk 3]
```diff
@@ -78,11 +78,11 @@ public void configure(String entityId, Consumer<String> onEditEvaluation) {
 			@Override
 			public void onSuccess(List<Evaluation> evaluations) {
 				for (Evaluation evaluation : evaluations) {
-					if(evaluation.getQuota() == null){
-						createEvaluationCardReactComponent(evaluation,timeInUtc,sessionToken,onEditEvaluation);
-					} else{
+//					if(evaluation.getQuota() == null){
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/evaluation/AdministerEvaluationsList.java]: can delete commented out code

**Review by jay-hodgson — APPROVED** [review]
Given that new Evaluations will automatically utilize the rounds editor, I agree with this self-serve migration path design.  Nice work!

**Pull request merged**